src/EditorFeatures/CSharpTest/ExtractMethod/ExtractMethodTests.cs

```diff
@@ -11182,6 +11182,45 @@ private static bool NewMethod()
     {
         return true;
     }
+}";
+            await TestExtractMethodAsync(code, expected);
+        }
+
+        [WorkItem(44260, "https://github.com/dotnet/roslyn/issues/44260")]
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task TopLevelStatement_ValueInAssignment()
+        {
+            var code = @"
+bool local;
+local = [|true|];
+";
+            var expected = @"
+bool local;
+local = NewMethod();
+
+bool NewMethod()
+{
+    return true;
+}";
+            await TestExtractMethodAsync(code, expected);
+        }
+
+        [WorkItem(44260, "https://github.com/dotnet/roslyn/issues/44260")]
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task TopLevelStatement_ArgumentInInvocation()
+        {
+            // Note: the cast should be simplified 
+            // https://github.com/dotnet/roslyn/issues/44260
```

Comment:
this issue doesn't seem related.

Reply:
The cast gets correctly simplified for similar situation inside in a method, so the problem may be specific to top-level statement somehow.